webpack/assets/javascripts/foreman_tools.js

```diff
@@ -103,43 +103,23 @@ export function initTypeAheadSelect(input) {
 // handle table updates via turoblinks
 export function updateTable(element) {
   const uri = new URI(window.location.href);
-  let searchTerm;
-  let perPage;
-  let isSearchForm;
-  let pageNum;
 
-  if (element !== undefined) {
-    isSearchForm = element.id === 'search-form';
+  const values = { };
 
-    pageNum = $('#cur_page_num').val();
-    if (pageNum !== undefined) {
-      uri.setSearch('page', pageNum);
-    }
-
-    if (isSearchForm || element.id === 'per_page') {
-      uri.setSearch('page', 1);
-    }
-
-    if (isSearchForm) {
-      searchTerm = $(element)
-        .find('.autocomplete-input')
-        .val();
-      if (searchTerm) {
-        uri.setSearch('search', searchTerm.trim());
-      }
-    }
+  values.page = $('#cur_page_num').val();
+  if (['per_page', 'search-form'].includes(element.id)) {
+    values.page = '1';
+  }
 
-    perPage = $('#per_page').val();
-    if (
-      perPage !== undefined &&
-      $('#search-form')
-        .find('.autocomplete-input')
-        .val() === undefined
-    ) {
-      uri.removeSearch('search');
-    }
-    uri.setSearch('per_page', perPage);
+  const searchTerm = $(element).find('.autocomplete-input').val();
+  if (searchTerm !== undefined) {
+    values.search = searchTerm.trim();
+    // if search term is empty set it to undefined so `uri.setSearch` will remove it
+    // from the search params if exists.
+    if (values.search === '') { values.search = undefined; }
```

Comment:
Are you sure this is needed? I tried playing around with it a bit and looks like it works fine even if the search is blank.

Reply:
It does work, but the "search" param is still shown in URI.
I don't mind remove this if-condition, though.